vncdotool/api.py

```diff
@@ -82,32 +81,37 @@ def disconnector(protocol: VNCDoToolClient) -> None:
     def __getattr__(self, attr: str) -> Any:
         method = getattr(self.factory.protocol, attr)
 
-        def errback(reason: Failure, *args: Any, **kwargs: Any) -> None:
-            self.queue.put(Failure(reason))
-
-        def callback(protocol: VNCDoToolClient, *args: Any, **kwargs: Any) -> Deferred:
+        def threaded_call(protocol: VNCDoToolClient, *args: Any, **kwargs: Any) -> Deferred:
             def result_callback(result: V) -> V:
                 self.queue.put(result)
                 return result
             d = maybeDeferred(method, protocol, *args, **kwargs)
             d.addBoth(result_callback)
             return d
 
-        def proxy_call(*args: Any, **kwargs: Any) -> Any:
-            reactor.callFromThread(self.factory.deferred.addCallbacks,
-                                   callback, errback, args, kwargs)
+        def errback_not_connected(reason: Failure, *args: Any, **kwargs: Any) -> None:
+            self.queue.put(reason)
+            return reason
```

Comment:
The type annotation does not match the implementation: ~Either delete the `return reason` to return `None` or~ change the signature to be
```suggestion
        def errback_not_connected(reason: Failure, *args: Any, **kwargs: Any) -> Failure:
            self.queue.put(reason)
            return reason
```
According to [addCallback](https://docs.twistedmatrix.com/en/stable/api/twisted.internet.defer.Deferred.html#addCallbacks) the function should return `Failure`.